src/client/entity-editor/relationship-editor/relationship-editor.js

```diff
@@ -0,0 +1,339 @@
+		}
+
+		return 100;
+	}
+
+	renderEntityTypeSelect() {
+		return (
+			<FormGroup>
+				<ControlLabel>Entity Type</ControlLabel>
+				<ReactSelect
+					name="entityType"
+					options={ENTITY_TYPE_VALUES}
+					value={this.state.entityType}
+					onChange={this.handleEntityTypeChange}
+				/>
+			</FormGroup>
+		);
+	}
+
+	renderEntitySelect() {
+		return (
+			<EntitySearchField
+				cache={false}
+				disabled={!this.state.entityType}
+				instanceId="publication"
+				label={_.get(this.state, 'entityType.label', 'Entity')}
+				name="entity"
+				type={_.get(this.state, 'entityType.id')}
+				value={this.state.targetEntity}
+				onChange={this.handleEntityChange}
+			/>
+		);
+	}
+
+	renderRelationshipSelect() {
+		const {baseEntity, relationshipTypes} = this.props;
+
+		const otherEntity = {
+			bbid: _.get(this.state, ['targetEntity', 'id']),
+			defaultAlias: {
+				name: _.get(this.state, ['targetEntity', 'text'])
+			},
+			type: _.get(this.state, ['targetEntity', 'type'])
+		};
+
+		const relationships = generateRelationshipSelection(
+			relationshipTypes, baseEntity, otherEntity
+		);
+
+		return (
+			<FormGroup>
+				<ControlLabel>Relationship</ControlLabel>
+				<ReactSelect
+					disabled={!this.state.targetEntity}
+					name="relationshipType"
+					optionRenderer={Relationship}
+					options={relationships}
+					value={this.state.relationship}
+					valueKey="selectId"
+					valueRenderer={Relationship}
+					onChange={this.handleRelationshipTypeChange}
+				/>
+			</FormGroup>
+		);
+	}
+
+	render() {
+		const {onCancel, onClose, baseEntity} = this.props;
+
+		const submitDisabled = this.calculateProgressAmount() < 100;
+
+		return (
+			<Modal show bsSize="large" onHide={onClose}>
+				<Modal.Header>
+					<Modal.Title>Add a relationship</Modal.Title>
+				</Modal.Header>
+				<Modal.Body>
+					<p>
+						<strong>
+							Use this form to add links between this{' '}
+							{baseEntity.type}
+							{' '}and other entities.
+						</strong>
+						{' '}For example, you can link a creator
+						to a work as an author, or a work to another work
+						to show translation or derivation.
+					</p>
+					<hr/>
+					<Row>
+						<Col md={10} mdOffset={1}>
+							<form>
+								<Row>
+									<Col md={4}>
+										{this.renderEntityTypeSelect()}
+									</Col>
+									<Col md={8}>
+										{this.renderEntitySelect()}
+									</Col>
+								</Row>
+								<Row>
+									<Col md={12}>
+										{this.renderRelationshipSelect()}
+									</Col>
+								</Row>
+							</form>
+						</Col>
+					</Row>
+				</Modal.Body>
+				<Modal.Footer>
+					<Row>
+						<Col md={10} mdOffset={1}>
+							<ProgressBar
+								bsStyle="success"
+								now={this.calculateProgressAmount()}
+							/>
+						</Col>
+					</Row>
+					<Button bsStyle="danger" onClick={onCancel}>Cancel</Button>
+					<Button
+						bsStyle="success"
+						disabled={submitDisabled}
+						onClick={this.handleSave}
+					>
+						Save
```

Comment:
A bit of a detail, but I'd suggest the button say "Add" or "Confirm" or something along those lines rather than "Save", as clicking it does not in fact save the relationship directly and you have to submit the changes.